packages/flutter/lib/src/material/carousel.dart

```diff
@@ -594,7 +616,11 @@ class _CarouselViewState extends State<CarouselView> {
 
   Widget _buildSliverCarousel(ThemeData theme) {
     // Determine the child count and builder based on whether we're using lazy loading
-    final int? childCount = widget.itemBuilder != null ? widget.itemCount : widget.children.length;
+    final int? childCount = widget.infinite
+        ? null
```

Comment:
In the API doc, we can also mention itemBuilder and itemCount will be ignored when infinite is not true:)

Reply:
Update API doc